main.go

```diff
@@ -81,21 +93,50 @@ func serverFinalizer(server *Server) {
 //
 // FIXME Pass in a list of 8-bit tags defining epochs.
 // The instance will generate its own secret key.
-func NewServer() (*Server, error) {
+func NewServer() (*Server, *string, error) {
 	// FIXME should we runtime.LockOSThread() here?
 	raw := C.randomness_server_create()
 	if raw == nil {
-		return nil, errors.New("failed to create randomness server")
+		return nil, nil, errors.New("failed to create randomness server")
 	}
 	server := &Server{raw: raw}
 	runtime.SetFinalizer(server, serverFinalizer)
-	return server, nil
+
+	var pkOutput [16384]byte
+	pkSize := C.randomness_server_get_public_key(
+		server.raw, (*C.uint8_t)(unsafe.Pointer(&pkOutput[0])))
+	if pkSize == 0 {
+		return nil, nil, errors.New("failed to get public key")
+	}
+	pk := base64.StdEncoding.EncodeToString(pkOutput[:pkSize])
+
+	return server, &pk, nil
+}
+
+// getEpoch returns the current epoch for the given timestamp.
+func getEpoch(t time.Time) (epoch, int64) {
+	currentMillisSinceFirstEpoch := t.UnixMilli() - firstEpochTimestampMs
+	epochsSinceFirstEpoch := currentMillisSinceFirstEpoch / epochLengthTimeMs
+	nextEpochTimeMs := firstEpochTimestampMs +
+		(epochLengthTimeMs * (epochsSinceFirstEpoch + 1))
```

Comment:
Because of leap seconds, this will drift relative to the civil calendar, but at least for P3A we're not planning on using epochs shorter than an hour, so I don't expect it to matter.